app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -80,10 +77,14 @@ class FxaRepo(
     private val _accountState: BehaviorSubject<AccountState> = BehaviorSubject.createDefault(NotAuthenticated)
     val accountState: Observable<AccountState> = _accountState.hide()
 
-    val receivedTabs: Observable<ReceivedTabs> = admIntegration.receivedTabsRaw
-        .mapToReceivedTabs()
-        .filterInvalidTabs(sentryIntegration)
+    val receivedTabs: Observable<FxaReceivedTab> = admIntegration.receivedTabsRaw
         .doOnNext { telemetryIntegration.receivedTabEvent(it) }
+        .filterMapToDomainObject()
+        .doOnNext { lastReceivedTab = it }
+
+    private var lastReceivedTab: FxaReceivedTab? = null
+    var queuedTab: FxaReceivedTab? = null
+        private set
```

Comment:
This approach is simple, but otherwise not the best.  I'm open to suggestions here.

Reply:
This is the solution I'd expect: when we receive a tab in the background, add it to a queue. When the app opens, pop the queue. It might feel weird because it's a nullable, mutable value but I'm not sure.